### ISSUE TITLE
Third payment processor

### ISSUE THREAD

**Issue by Changaco** [liberapay/liberapay.com#1394]
We didn't actually have an issue about this, so I'm creating one. Liberapay currently supports Stripe and PayPal (barely), however Stripe only covers a relatively small number of countries (as a European citizen I especially dislike the fact that they don't operate in all EU members), and PayPal is hated and boycotted by some people, so it would be good to have a 3rd option.

**Comment by Changaco** [comment]
Relevant issues: #374 contains a long list of various kinds of payment processors, #1309 would probably be the best choice if it was operational.

**Comment by unfa** [comment]
I agree. Stripe doesn't operate in Poland, so my only option is PayPal.

How about [Transferwise](https://transferwise.com/)?

One of my contractors (a FLOSS developer) use them and I've heard they have very low fees, and also great currency exchange terms.

And they support Poland :D

**Comment by Changaco** [comment]
[TransferWise's API](https://api-docs.transferwise.com/) doesn't provide what we need (yet?).

**Comment by techtonik** [comment]
I've just discovered https://interledger.org/ - could be better than payment processor.

**Comment by zaynetro** [comment]
Following PSD2 directive EU banks should provide APIs for allowing initiating and tracking payments. Technically, it should be possible for 3rd parties to transfer money between accounts.

Example: https://developer.nordeaopenbanking.com/

Unfortunately, I am not knowledgeable in this topic nor am I aware of any payment provides that build on top of that. Also I fear that to build the integration yourself would require partnering with each individual bank.

Quick searching revealed a few potential payment providers: 

* https://www.figo.io/en/
* https://plaid.com/
* https://synapsefi.com/
* https://tink.com/

I don't know the details of the mentioned companies but decided to document the findings anyway. I might return and read more about those.

**Comment by mattbk** [comment]
Tweet from GoCardless: https://twitter.com/GoCardless/status/1108749224617934848

> Excited to share that we will be coming to the US soon

https://gocardless.com/en-us/

**Comment by Changaco** [comment]
GoCardless is a lower-level payment processor, we were using it through Mangopay but we can't use it directly. Stripe already supports ACH, but only for US platforms.

**Comment by Changaco** [comment]
@zaynetro It's true that the PSD2 directive may eventually allow us to support direct SEPA transfers from one donor to one recipient, but that's of limited usefulness to a platform like Liberapay.

**Comment by mattbk** [comment]
Does Gravity count as an option? https://gravitypayments.com/

**Comment by wushepeng** [comment]
> It's true that the PSD2 directive may eventually allow us to support direct SEPA transfers from one donor to one recipient, but that's of limited usefulness to a platform like Liberapay.

I would really like something like that as a possible method:

- You have a special liberapay bank account. I can send SEPA transfert to it with a one time code. When my transfert is processed, my liberapay account is credited and I can distribute the curency I have. Eventually, the currency is transformed in tokens.

- when someone wants to withdraw currency, you make a SEPA transfert to his bank account. You can add conditions like minimum amount.

Ideally, a method that can avoid banks would be the must for me, but I fear there is some legal thread about that. But if not, allow crytocoin (I specially think at Ǧ1) should be a good option.

**Comment by mpass3** [comment]
Mailbox.org just went with Adyen, which is a dutch payments processor. I would support a european payments processor instead of an American. We just had an issue in Canada where a "Cuban" sandwhich company had $14,000 withheld because the American clearer wouldn't release the funds due to their economic blocade on the country. It's imperialism and I don't think using american payment providers fits with the mission and values of Liberapay. 

https://mailbox.org/en/post/short-news-from-now-on-credit-card-payments-are-also-possible-at-mailbox-org

**Comment by revi** [comment]
@mpass3: [Adyen told us no](https://github.com/liberapay/liberapay.com/issues/374#issuecomment-407367797).

**Comment by mpass3** [comment]
> @mpass3: [Adyen told us no](https://github.com/liberapay/liberapay.com/issues/374#issuecomment-407367797).

I'm going to write an email to Adyen requesting they change their mind.

**Comment by mpass3** [comment]
I wrote the email requesting they change their mind and to contact me if they wanted to continue the discussion. I'll reply back here; Has the team considered: https://www.mollie.com/en/ ? It seems to be reasonably priced (1.8% is less than 2.9% for Paypal) and accepts most credit cards. They also have a recurring payment feature.

**Comment by Changaco** [comment]
The country of origin of a multinational company isn't very relevant. Adyen does business in the US, consequently they have to follow US regulations and can be pressured by the US government, just like Stripe and PayPal (who also have to follow EU regulations because they do business in the EU).

**Comment by Changaco** [comment]
As far as I can tell Mollie only supports a small number European countries, they would need to expand and improve a lot more in order to become an interesting alternative to Stripe.

**Comment by mpass3** [comment]
> As far as I can tell Mollie only supports a small number European countries, they would need to expand and improve a lot more in order to become an interesting alternative to Stripe.

Fair enough; I'm not an active contributor so my opinion should be taken with a grain of salt; I will say however that entering a relationship with a vendor while their are new and expanding can lead to exposure for both parties and better pricing in the future. I'd also interject that I can make a payment on Mollie with my Canadian credit card, as it does allow for non-EU payments;

To your point about the origin of the multinational not being relevant; I disagree. Adyen only has to follow USA regulation while operating in the USA. US-domiciled countries have to follow USA regulations regardless of where they operate. All companies can be pressured by the US government but the goal of this issue (IMO) is to minimize disruption and diversify payment processors as geopolitical tensions lead to economic disturbances. Freedom Fries could easily become Freedom Open Source projects in the future; and non-USA open source projects could be provided with headaches that could be resolved by using a european-domiciled payment processor. Respectfully.

**Comment by Changaco** [comment]
> I'd also interject that I can make a payment on Mollie with my Canadian credit card, as it does allow for non-EU payments;

The problem isn't the donor's country, it's the recipient's country. The creators who want to use Liberapay are all over the world, consequently the ideal payment processor for Liberapay would support paying out to creators worldwide. (Mangopay wasn't good at sending money outside the EU.)

**Comment by techtonik** [comment]
@Changaco it is impossible to find such payment processor, because every county is protective to its own payment processors. But it is possible to use country level payment processors.

1. Identify source country
2. Identify target country
3. Match payment provider
4. If payment provider is not found, provide an algorithm how to find and add it to Liberapay

**Comment by Changaco** [comment]
@techtonik It's not impossible for a payment processor to send money to almost any country in the world. In fact we already support the one payment processor that does that: PayPal. The problem is that PayPal is bad at a lot of other things.

**Comment by Changaco** [comment]
@unfa Stripe is available in Poland now, you should create your account and connect it to your Liberapay: https://liberapay.com/about/me/payment

**Comment by FediVideos** [comment]
> [TransferWise's API](https://api-docs.transferwise.com/) doesn't provide what we need (yet?).

Is the API any better now in 2021? Would TransferWise be possible do you think?

**Comment by Changaco** [comment]
@FediVideos Not really. It might be possible to integrate Wise to enable receiving manual bank transfers, but that payment method is such a mess that I'm not sure it's a good idea (see #2026).

**Comment by IzzySoft** [comment]
Going by that argument I wonder how people do their "regular" bank transfers :smile: Couldn't that be used at least for withdrawing? I guess us content creators are capable of copy-pasting our own IBAN into some form. That could be kept then with the other profile data, so we make no mistakes when copying it the next time :speak_no_evil:

Besides: I've received via Transferwise from the US as well as from other EU countries. Worked excellent each time, whether inside the EU or across the sea.

**Comment by Changaco** [comment]
@IzzySoft Three years after the Mangopay crisis you still don't understand that there's no such thing as withdrawing money from Liberapay?

---

To clarify, the Wise integration that might be possible would work like this:

1. Liberapay obtains API access to the recipient's Wise account (which means that Liberapay would see every payment to and from the account, even the ones that have nothing to do with Liberapay);
2. if a donor chooses to pay with a manual bank transfer, then Liberapay shows the recipient's bank account number to the donor (which means that the recipient would be exposed to fraudulent uses of this number), as well as a unique code to identify the payment (which some donors will fail to send along with the money, thus requiring manual reconciliation);
3. Liberapay monitors the Wise account to see if the payment arrives and notifies the donor when that happens or after 30 days if the payment hasn't arrived.

**Comment by IzzySoft** [comment]
Thanks for your response, Charly!

> Three years after the Mangopay crisis you still don't understand that there's no such thing as withdrawing money from Liberapay?

Sad, huh? No offense meant, I know how hard things can go, but this is indeed disappointing. Are you saying I should finally give up on Liberapay? Though it doesn't work any longer *for me* I'm still recommending it to others who can live with PayPal/Stripe (I can't, as pointed out multiple times). My account with you still exists, though nobody can donate to it. I meanwhile removed most links pointing to it, mainly leaving one ~striked-through~ line stating why it's currently broken for me.

Alternatives seem always impossible here – as if you were waiting for a magic processor working galaxy-wide you just had to "plug in" without any hazzle and it would work. Don't get me wrong, I very much appreciate what Liberapay does, but this point simply frustrates me.

To your points:

1. "the recipient's Wise account": What is that? I don't have such, but I received payments via Transferwise more than once, directly to my bank account. I didn't sign any paper to "give access". Private payments as well as "official" ones from OpenCollective. So obviously, this point is not needed. It's also one of the reasons I refuse Stripe: if I only want to receive, I see no reasons to give a 3rd party access to all my bank account data. When I have to pay a bill, I cannot monitor the recipient's account either to see whether the money really arrived there, so why should this be needed?
2. In my case, if a donor wishes so, the IBAN can be found on my website. With Liberapay acting as middle-man, I don't understand why it needed to show the final recipient's account details: if it's a direct transfer, no middle-man would be needed. If there's a middle-man, the transfer would go via their account and forwarded. In that case I'd of course need to let the middle-man know my IBAN – but that was never doubted.
3. I wonder what OpenCollective monitors then, as I don't have a Wise account.

I guess I'm falling into the same trap again which, from my naïve point of view, makes Liberapay more complicated than needed: Would you establish Transferwise, the only payments I could receive via it would be those the donors send in via Transferwise (as to receive PayPal donations I'd need a PayPal account). This is why people ask what for a middle-man is needed if it doesn't "bundle" those. I know there are reasons why you cannot do that, though I never fully understood them.

**P.S.:** before you attribute this to my "stubbornness", I'm not the only one having been cut-off from Liberapay because of "this". I know several other "privacy proponents" either not considering LP because of that, or having closed their existing accounts for those reasons. And please don't ask me again to "just try Stripe, it's not that bad" – that really hurt back then as it makes the impression my concerns either haven't been understood or, worse, are simply ignored.

**Comment by SolarDesalination** [comment]
Maybe we'll get a crypto payment processor in the future that will allow easier acceptance internationally without political-based sanctions and with privacy by design transfers.

**Comment by Changaco** [comment]
@IzzySoft Your comment is off-topic. Instead of trying to answer it directly, I'm going to respond to the following question, event though it's also off-topic: “As someone who wants to receive money, why can't I simply input my bank account number in Liberapay like I can in Open Collective?”

Because Open Collective has an additional middleman: the fiscal host. A payment to a collective is processed using the Stripe or PayPal account of the collective's fiscal host, and the money is stored in one of the host's Stripe/PayPal/Wise/bank accounts. When an expense is submitted and approved, the fiscal host pays out the stored money to the bank account of the person who submitted the expense. There's nothing “direct” about this, the money did go through Stripe or PayPal, but the final recipients don't see that because the fiscal host shields them from it. The reason why the hosts can do stuff that the platform can't is that the money they're holding and sending belongs to them, whereas the platform handles money that belongs to other people.

As stated in <https://liberapay.com/about/teams>, implementing fiscal hosting is part of Liberapay's long term plans, but don't hold your breadth. The relevant issues are #505 and #1108.

**Comment by IzzySoft** [comment]
Thanks Charly, and apologies for digressing. Nice short description on how OC works btw :smiley: And your last paragraph is one of the reasons I didn't yet give up on LP, though my account is rather "dead" at the moment. Keeping my fingers crossed for good progress – but as you suggested, I don't expect anything soon. Open for positive surprises, though :smile: Withdrawing myself from the discussion again then with a "heads up": there are more folks out there waiting to re-join. Keep up the good work! And I hope a fitting payment processor can be found in the near future.

**Comment by yoandrygc** [comment]
Hello, is it possible to add Tropipay.com? It is based on Spain. It has an API.

"Tropipay is an electronic wallet that allows you to execute the most common financial operations of the day to day. It is easy and safe.

Our industry-leading technology protects your money and ensures it reaches its destination safely."

*Note: I am not an employee. I have an account, and would be very useful, for me and many others, to use it in conjunction with Liberapay to boost collaboration to open-source projects.

**Comment by Altonss** [comment]
Maybe https://www.fintecture.com/ might be a thing? I tried it once and you can do a payment without a credit card (you can also pay by card), just by initiating a bank transfer (you are redirected to your bank website)...
Their SDK is available here https://github.com/Fintecture

**Comment by ysfchn** [comment]
Apologies if this is the wrong place to talk about, however I couldn't find a more suitable issue.

I'm from Turkey, and as you can probably guess, both PayPal and Stripe (and a lot of payment services, due to strict requirements in here) are not available in Turkey. I came across to this comment:

> It might be possible to integrate Wise to enable receiving manual bank transfers, but that payment method is such a mess that I'm not sure it's a good idea.

I'm aware as Wise is not a "payment processor" they mostly "hold and transfer money", and it is not straight-forward as much as other methods, but however, it would be very helpful for someone like me who can't use most of global payment processors and doesn't have a choice to receive donations globally and directly. (even Wise operates in Turkey by a middleman)

FYI, Wise have an example [image](https://wise.com/public-resources/assets/api-docs/bank_example.png) in their API docs about how flow works.

**Comment by MagicLike** [comment]
Any updates on Mollie? I have read through their Website and it sounds pretty good...

**Comment by Changaco** [comment]
Mollie is only available to businesses and nonprofits based in the [EEA](https://en.wikipedia.org/wiki/European_Economic_Area), Switzerland and the UK ([source](https://help.mollie.com/hc/en-us/articles/115002116105-Can-I-use-Mollie-s-services-in-my-country-)). Those countries are almost all covered by Stripe already, so adding Mollie isn't interesting enough to be worth the effort.

**Comment by MagicLike** [comment]
Is there a european payment processor, which is worth implementing?
Because I don't really trust American companys with payments (especially PayPal).

**Comment by SolarDesalination** [comment]
I find it pretty ironic @Changaco that you write that there is no point to use a European payments provider because there is an equivalent American payments provider. I really don't see a purpose for Liberapay because there is already OpenCollective, Patreon, and PayPal, so using LiberaPay isn't interesting enough to be worth the effort..

**Comment by MagicLike** [comment]
I also think that it is important to have a European equivalent - as I said - for privacy.
And that is something Liberapay should identify with.

**Comment by cadadr** [comment]
If a new payment method is added that does not cater to countries that aren't already catered to, especially if a payment processor that excludes _developing_ and _underdeveloped_ countries, LiberaPay will pretty much go off my radar.

Using Patreon, Ko-Fi etc., and also LiberaPay, in the global south is extremely cumbersome, with paying up significant cuts to intermediaries like Wise and WU pretty much mandatory.

Maybe the Europeans who are already well served should quit whining?

**Comment by VincentSC** [comment]
Why is this still an issue after many years? Does Liberapay get paid by Stripe?

**Comment by clehner** [comment]
Integrating a new payment processor using only free/libre/open-source software could address #1279 (as an alternative) and unblock #1374. (Hopefully such a  processor would increase coverage of recipient/creator countries/currencies as well)

**Comment by Fuseteam** [comment]
> [TransferWise's API](https://api-docs.transferwise.com/) doesn't provide what we need (yet?).

TransferWise is now wise, what do we need from a payment processor? it has been 7 years since https://github.com/liberapay/liberapay.com/issues/619 perhaps they do provide what is needed now?

the api docs are now at https://docs.wise.com/api-reference

**Comment by zuckerruebe** [comment]
Also, considering that the US/its president has gone rogue, I think it might be interesting to add a European payment processor or any other kind of "European option".

**Comment by Fuseteam** [comment]
would love to know what is needed here so i can also look into if wise has a support for it— or https://www.mollie.com/ or https://paddle.com for that matter

**Comment by h7p3ri0n** [comment]
Hello @zuckerruebe! I am currently implementing [Taler](https://github.com/liberapay/liberapay.com/issues/1309), which allows for a far better level of privacy than Paypal and Stripe. Furthermore, we might consider integrating some services provided by [Adyen](https://www.adyen.com/) in the future. Did you have a specific provider in mind? Have a good day!

**Comment by zuckerruebe** [comment]
> Hello [@zuckerruebe](https://github.com/zuckerruebe)! I am currently implementing [Taler](https://github.com/liberapay/liberapay.com/issues/1309), which allows for a far better level of privacy than Paypal and Stripe. Furthermore, we might consider integrating some services provided by [Adyen](https://www.adyen.com/) in the future. Did you have a specific provider in mind? Have a good day!

No, not really, don't really know what I'm talking about. :)
